src/Facility.Core/Http/HttpClientService.cs

```diff
@@ -101,8 +103,66 @@ protected HttpClientService(HttpClientServiceSettings? settings, Uri? defaultBas
 			{
 				var contentType = mapping.RequestBodyContentType ?? requestHeaders?.GetContentType();
 				httpRequest.Content = GetHttpContentSerializer(requestBody.GetType()).CreateHttpContent(requestBody, contentType);
-				if (m_disableChunkedTransfer)
+				if (m_shouldCompressRequest(request))
+					httpRequest.Content = await CompressContentAsync(httpRequest.Content, cancellationToken).ConfigureAwait(false);
+				else if (m_disableChunkedTransfer)
 					await httpRequest.Content.LoadIntoBufferAsync().ConfigureAwait(false);
+
+				async static Task<HttpContent> CompressContentAsync(HttpContent httpContent, CancellationToken cancellationToken)
+				{
+					Stream? contentStream = null;
+					Stream? compressedContentStream = null;
+
+					try
+					{
+						// copy the existing HTTP content into a memory stream
```

Comment:
Obviously it would preclude optionally not compressing, but would there be an advantage to compressing "live"?

Reply:
Presumably it would be more efficient to write the gzip output directly to the HTTP request (especially for large requests), rather than buffering it all in memory, but does commit one to sending the gzipped payload, even if it's larger.

Reply:
Given that running on a container with less memory can be advantageous, I like the idea of streaming straight from the DTO rather than through two memory buffers. "Disable chunked transfer" would still buffer it into memory, and better, since it would use the array-pooled internal implementation.